### ISSUE TITLE
Candlestick charts in R

### ISSUE THREAD

**Issue by debsush** [ropensci/plotly#630]
Hi,

I am new to plotly and was wondering if the following is doable with the package:
1. Create a candle stick natively as done in the python example below:
import plotly.plotly as py
from plotly.tools import FigureFactory as FF
from datetime import datetime

import pandas.io.data as web

df = web.DataReader("aapl", 'yahoo', datetime(2007, 10, 1), datetime(2009, 4, 1))
fig = FF.create_candlestick(df.Open, df.High, df.Low, df.Close, dates=df.index)
py.iplot(fig, filename='finance/aapl-candlestick', validate=False)

2. Just like ggplot, can we extend this to plot.xts?
3. Can we make the ChartSeries function from quantmod interactive with plotly?

Even if it is still not in place, was wondering if you could guide me as to how can I start developing these R charting functions within plotly?

Regards,
SD

**Comment by jackparmer** [comment]
Hey @debsush -

Have you seen this?
http://moderndata.plot.ly/candlestick-charts-using-plotly-and-quantmod/

![image](https://cloud.githubusercontent.com/assets/1865834/16217645/369768a8-372a-11e6-8652-50efb611ac64.png)

It might help to get you started.

**Comment by debsush** [comment]
Thanks Jack.

I have seen the implementation - it is definitely a good start.  While there is a ton of work to be done on this chart which I can take up on my own, what bothers me is the render speed.

TODO:
1. For 10 yrs and more daily data, feature to aggregate daily data into weekly and monthly data etc.
2. When zoomed-in on daily data, the current chart continues to show date in MMM YYYY format etc.

All of the above are very much doable but I am a bit concerned about the render speed.  Currently a 10 yr daily OHLC data takes 8-12 seconds to render.  Compare this with ChartSeries in quantmod, render time is in milliseconds which is so cool.

@timelyportfolio Your feedback will be most welcome on this.

Regards
SD

**Comment by cpsievert** [comment]
When #628 is finished, you'll be able to do something like:

```r
library(plotly)
library(quantmod)
library(dplyr)

prices <- getSymbols("MSFT", auto.assign = F)
dat <- as.data.frame(prices)
dat$date <- index(prices)
names(dat) <- sub("^MSFT\\.", "", names(dat))
# create variable with hovertext displaying values of every variable
vars <- Map(function(x, y) paste0(x, ": ", y), names(dat), dat)
dat$txt <- Reduce(function(x, y) paste0(x, "<br />", y), vars)

plot_ly(dat, x = ~date, xend = ~date, color = ~Close > Open, 
        colors = c("red", "forestgreen"), hoverinfo = "none") %>%
  add_segments(y = ~Low, yend = ~High, line = list(width = 1)) %>%
  add_segments(y = ~Open, yend = ~Close, line = list(width = 3)) %>%
  add_markers(y = ~(Close + Open)/2, text = ~txt, hoverinfo = "text", opacity = 0) %>%
  toWebGL() %>%
  layout(showlegend = FALSE, yaxis = list(title = "Price")) 
```

**Comment by cpsievert** [comment]
Closed via d2bf2a1eb56d824a091e2cc9afea09a5968dd050.

**Comment by GoChartingAdmin** [comment]
I am getting the below error on running your code

Error in function_list[[i]](value) : 
  could not find function "add_segments"
